group-coordinator/src/main/java/org/apache/kafka/coordinator/group/assignor/TopicIds.java

```
@@ -0,0 +1,176 @@
+    ) {
+        this.topicNames = topicNames;
+        this.image = image;
+    }
+
+    @Override
+    public int size() {
+        return topicNames.size();
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return topicNames.isEmpty();
+    }
+
+    @Override
+    public boolean contains(Object o) {
+        if (o instanceof Uuid) {
+            Uuid topicId = (Uuid) o;
+            TopicImage topicImage = image.getTopic(topicId);
+            if (topicImage == null) return false;
+            return topicNames.contains(topicImage.name());
+        }
+        return false;
+    }
+
+    private static class TopicIdIterator implements Iterator<Uuid> {
+        final Iterator<String> iterator;
+        final TopicsImage image;
+        private Uuid next = null;
+
+        private TopicIdIterator(
+            Iterator<String> iterator,
+            TopicsImage image
+        ) {
+            this.iterator = iterator;
+            this.image = image;
+        }
+
+        @Override
+        public boolean hasNext() {
+            if (next != null) return true;
+            Uuid result = null;
+            do {
+                if (!iterator.hasNext()) {
+                    return false;
+                }
+                String next = iterator.next();
+                TopicImage topicImage = image.getTopic(next);
+                if (topicImage != null) {
+                    result = topicImage.id();
+                }
+            } while (result == null);
+            next = result;
+            return true;
+        }
+
+        @Override
+        public Uuid next() {
+            if (!hasNext()) throw new NoSuchElementException();
+            Uuid result = next;
+            next = null;
+            return result;
+        }
+    }
+
+    @Override
+    public Iterator<Uuid> iterator() {
+        return new TopicIdIterator(topicNames.iterator(), image);
+    }
+
+    @Override
+    public Object[] toArray() {
+        return new Object[0];
```

Should we implement this one? Otherwise, we should throw `UnsupportedOperationException`.